packages/core/src/config/order/order-by-code-access-strategy.ts

```diff
@@ -0,0 +1,65 @@
+import ms from 'ms';
+
+import { RequestContext } from '../../api/common/request-context';
+import { InjectableStrategy } from '../../common/types/injectable-strategy';
+import { Order } from '../../entity/order/order.entity';
+
+/**
+ * @description
+ * The OrderByCodeAccessStrategy determines how access to a placed Order via the
+ * orderByCode query is granted.
+ * With a custom strategy anonymous access could be made permanent or tied to specific
+ * conditions like IP range or an Order status.
+ *
+ * @example
+ * This example grants access to the requested Order to anyone – unless it's Monday.
+ * ```TypeScript
+ * export class NotMondayOrderByCodeAccessStrategy implements OrderByCodeAccessStrategy {
+ *     canAccessOrder(ctx: RequestContext, order: Order): boolean {
+ *         const MONDAY = 1;
+ *         const today = (new Date()).getDay();
+ *
+ *         return today !== MONDAY;
+ *     }
+ * }
+ * ```
+ *
+ * @docsCategory orders
+ * @docsPage OrderByCodeAccessStrategy
+ */
+export interface OrderByCodeAccessStrategy extends InjectableStrategy {
+    /**
+     * @description
+     * Gives or denies permission to access the requested Order
+     */
+    canAccessOrder(ctx: RequestContext, order: Order): boolean | Promise<boolean>;
+}
+
+/**
+ * @description
+ * The default OrderByCodeAccessStrategy used by Vendure. It permitts permanent access to
+ * the Customer owning the Order and anyone within a given time period after placing the Order
+ * (defaults to 2h).
+ *
+ * @param anonymousAccessDuration value for [ms](https://github.com/vercel/ms), e.g. `2h` for 2 hours or `5d` for 5 days
+ *
+ * @docsCategory orders
+ * @docsPage OrderByCodeAccessStrategy
+ */
+export class DefaultOrderByCodeAccessStrategy implements OrderByCodeAccessStrategy {
+    canAccessOrder(ctx: RequestContext, order: Order, anonymousAccessDuration: string = '2h'): boolean {
```

Comment:
The `anonymousAccessDuration` argument should be passed to the strategy constructor, not to the `canAccessOrder` method. In this way, the developer will be able to use it like:

```ts
orderByCodeAccessStrategy: new DefaultOrderByCodeAccessStrategy('5h'),
```

Reply:
Oopsie, fixed. That's a bit embarrassing.